internal/provider/resource_silo_test.go

```diff
@@ -0,0 +1,219 @@
+)
+
+type resourceSiloConfig struct {
+	BlockName            string
+	SiloName             string
+	TLSCertificateBase64 string
+	TLSPrivateKeyBase64  string
+}
+
+var resourceSiloConfigTpl = `
+resource "oxide_silo" "{{.BlockName}}" {
+  name          = "{{.SiloName}}"
+  description   = "Managed by Terraform."
+  discoverable  = true
+  identity_mode = "local_only"
+
+  quotas = {
+    cpus    = 2
+    memory  = 8589934592
+    storage = 8589934592
+  }
+
+  mapped_fleet_roles = {
+    admin  = ["admin", "collaborator"]
+    viewer = ["viewer"]
+  }
+
+  tls_certificates = [
+    {
+      name        = "self-signed-wildcard"
+      description = "Self-signed wildcard certificate for *.sys.r3.oxide-preview.com."
+      cert        = base64decode("{{.TLSCertificateBase64}}")
+      key         = base64decode("{{.TLSPrivateKeyBase64}}")
+      service     = "external_api"
+    },
+  ]
+
+  timeouts = {
+    create = "1m"
+    read   = "2m"
+    update = "3m"
+    delete = "4m"
+  }
+}
+`
+
+var resourceSiloUpdateConfigTpl = `
+resource "oxide_silo" "{{.BlockName}}" {
+  name          = "{{.SiloName}}"
+  description   = "Managed by Terraform."
+  discoverable  = true
+  identity_mode = "local_only"
+
+  quotas = {
+    cpus    = 4           # 2 -> 4
+    memory  = 17179869184 # 8 GiB -> 16 GiB
+    storage = 17179869184 # 8 GiB -> 16 GiB
+  }
+
+  mapped_fleet_roles = {
+    admin  = ["admin", "collaborator"]
+    viewer = ["viewer"]
+  }
+
+  tls_certificates = [
+    {
+      name        = "self-signed-wildcard"
+      description = "Self-signed wildcard certificate for *.sys.r3.oxide-preview.com."
+      cert        = base64decode("{{.TLSCertificateBase64}}")
+      key         = base64decode("{{.TLSPrivateKeyBase64}}")
+      service     = "external_api"
+    },
+  ]
+}
+`
+
+func TestAccCloudResourceSilo_full(t *testing.T) {
```

Comment:
The prefix `TestAccCloud` is reserved for "cloud" resources that we run as a on every release against the colo rack. 

```suggestion
func TestAccSiloResourceSilo_full(t *testing.T) {
```